To-do list/Controllers/LoginViewController.swift

```diff
@@ -23,29 +26,86 @@ class LoginViewController: UIViewController {
         if passwordTextField.text?.isEmpty == true {
             return "Please enter your password"
         }
+        else if passwordTextField.text!.count < 8 {
+            return "Password must have at least 8 characters"
+        }
         return nil
     }
     
-    @IBAction func signInButton(_ sender: UIButton) {        
+    private func validateCredentials() -> Bool {
         let loginMessage = validateEmail()
         let passMessage = validatePass()
         
-        if loginMessage == nil && passMessage == nil {
-            let taskListVC = TaskListViewController.loadFromStoryboard(type: TaskListViewController.self)
-            self.navigationController?.pushViewController(taskListVC, animated: true)
-        } else {
+        if loginMessage != nil || passMessage != nil {
             var message = String()
             if let loginMessage = loginMessage {
                 message = loginMessage
             } else if let passMessage = passMessage {
                 message = passMessage
             }
             Alert.showBasic(title: message, vc: self)
+            return false
+        } else {
+            return true
+        }
+    }
+    
+    @IBAction func signInButton(_ sender: UIButton) {
+        if validateCredentials() {
+            guard
+                let email = loginTextField.text,
+                let password = passwordTextField.text
+            else {return}
+            
+            authApiHelper.executeLoginRequest(email: email, password: password) {[weak self] result in
+                guard let self = self else {
+                    return
+                }
+                var message = String()
```

Comment:
add 1 empty line after `var message = String()`, because next logical block is - `if/else flow control`